### PR TITLE
MM-24505-Update-help-text-leave-team-modal

### PR DESCRIPTION
**Pull request by sridhar02**
#### Summary

Update help text in Leave Team modal

#### Ticket Link

  Fixes https://github.com/mattermost/mattermost-server/issues/14401

#### Screenshots

image

### CODE REVIEW

**Comment by sridhar02** [comment]
I am not sure how to get the dynamic values of the public and private channels, can anyone suggest me how to get and I think as we have `currentTeamId` we will be able to get some list from that but I am not sure from which method we have to use to get those?

**Comment by hanzei** [comment]
@hmhealey Can you help with the question above? :point_up:

**Comment by hmhealey** [comment]
@sridhar02 If you take a look at `components/leave_team_modal/index.ts`, you'll see the `mapStateToProps` function that's used to get parts of the app state (like the current team ID currently) and pass them into `LeaveTeamModal` as props. The user's channels will already have been loaded in the redux state, so you'll have to pull those numbers out of these using a selector like the one you made that gets channels for the quick switcher.

In particular, you'll probably need one that looks at `state.entities.channels.channels` (all channels loaded by the client) and `state.entities.channels.myMembers` (the ChannelMember objects representing the channels that the current user is a member of) to count how many channels of each type that the user is a member of on the current team. If you need any help doing that, I'd be glad to help out more.

**Comment by sridhar02** [comment]
@hanzei @hmhealey can you review the PR now?

**Review by esethna — CHANGES_REQUESTED** [review]
Thanks @sridhar02, if there are no public or private channels that the user is a member of we should not mentioned it, for example:

```
You will be removed from all 3 public channels on this team
You will be removed from all 3 private channels on this team
```
image

File: i18n/en.json [hunk 1]
```diff
@@ -2984,7 +2984,7 @@
   "leave_private_channel_modal.leave": "Yes, leave channel",
   "leave_private_channel_modal.message": "Are you sure you wish to leave the private channel {channel}? You must be re-invited in order to re-join this channel in the future.",
   "leave_private_channel_modal.title": "Leave Private Channel {channel}",
-  "leave_team_modal.desc": "You will be removed from all public and private channels.  If the team is private you will not be able to rejoin the team.  Are you sure?",
+  "leave_team_modal.desc": "**You will be removed from all {num_of_public_channels} public channels and {num_of_private_channels} private channels on this team**. If the team is private you will not be able to rejoin the team without an invitation. Are you sure?",
```
Comment (hmhealey) [inline i18n/en.json:2987]: This won't be grammatically correct when the user is a member of exactly 1 channel of a given type, but you can use the `{plural}` keyword to automatically pluralize words based on the number of public/private channels in the team. There's information on using that [here](https://formatjs.io/docs/core-concepts/icu-syntax#plural-format), but you'll want to write something like
```
{numOfPublicChannels, plural, one {channel} other {channels}}
```
This results in the word "channel" if `numOfPublicChannels` is equal to 1 or "channels" if it is any other number.

**Review by hmhealey — CHANGES_REQUESTED** [review]

**Review by hmhealey — CHANGES_REQUESTED** [review]
Also, could you add separate strings for if the user isn't a part of any public or private channels like "You will be removed from all X public channels" and "You will be removed from all X private channels"? I don't think there will be any cases where the user isn't a member of any channels of either type, so I don't think we need to handle that case.

**Comment by esethna** [comment]
@sridhar02 let us know if you need help with the above review comments?

**Comment by sridhar02** [comment]
@hmhealey @esethna @sudheerDev I have corrected with necessary conditions but getting an i18n check error can you help with it?

**Comment by sudheerDev** [comment]
@sridhar02 Hey, you need to run i18 tool to fix this, Docs can be found at https://developers.mattermost.com/contribute/webapp/using-i18n-extract/. Let me know if you want me to add a commit to fix it.

File: components/leave_team_modal/leave_team_modal.tsx [hunk 2]
```diff
@@ -40,11 +46,48 @@ export default class LeaveTeamModal extends React.PureComponent<Props> {
...
+        );
         this.props.actions.toggleSideBarRightMenu();
     };
 
     render() {
+        const numOfPublicChannels = this.props.publicChannels.length;
+        const numOfPrivateChannels = this.props.privateChannels.length;
+        const zero = 0;
+        let modalMessage;
+        if (numOfPublicChannels !== 0 && numOfPrivateChannels !== 0) {
```
Comment (sudheerDev) [inline components/leave_team_modal/leave_team_modal.tsx:61]: defining a const for zero seemed like a little odd, but if we want to go ahead you might as well replace the other instances here

**Review by sudheerDev — CHANGES_REQUESTED** [review]

File: components/leave_team_modal/leave_team_modal.tsx (hunk 2 above)
Comment (sridhar02) [reply components/leave_team_modal/leave_team_modal.tsx:60]: @sudheerDev  I have removed the constant for now keeping the conditions check to zero. If you want to change do let me know. I have also corrected the i18n error by running the script.

**Review by sudheerDev — APPROVED** [review]
LGTM, Thank you @sridhar02

**Review by esethna — APPROVED** [review]

**Comment by jgilliam17** [comment]
Thank you @sridhar02 
One small suggestion 
@esethna 
We should remove `all` since we are already displaying the channel count.
Looks a bit off when member/guest is leaving the team with 1 channel > `all 1 public channel`
Screen Shot 2020-10-16 at 10 44 17 AM

**Comment by jgilliam17** [comment]
One more thing - maybe we should also add a check if logged in user is a guest - guests should still see the longer warning when leaving any team, public or private, neither can be joined without an invitation

**Comment by esethna** [comment]
@jgilliam17 thanks, agree with your suggestions ^

File: i18n/en.json [hunk 3]
```diff
@@ -3041,7 +3041,9 @@
   "leave_private_channel_modal.leave": "Yes, leave channel",
   "leave_private_channel_modal.message": "Are you sure you wish to leave the private channel {channel}? You must be re-invited in order to re-join this channel in the future.",
   "leave_private_channel_modal.title": "Leave Private Channel {channel}",
-  "leave_team_modal.desc": "You will be removed from all public and private channels.  If the team is private you will not be able to rejoin the team.  Are you sure?",
+  "leave_team_modal_private.desc": "**You will be removed from all {num_of_private_channels} private { num_of_private_channels,plural,one {channel} other {channels} } on this team**. If the team is private you will not be able to rejoin the team without an invitation. Are you sure?",
```
Comment (jgilliam17) [inline i18n/en.json:3044]:
```suggestion
  "leave_team_modal_private.desc": "**You will be removed from {num_of_private_channels} private { num_of_private_channels,plural,one {channel} other {channels} } on this team.** If the team is private you will not be able to rejoin the team without an invitation. Are you sure?",
```

File: i18n/en.json [hunk 4]
```diff
@@ -3041,7 +3041,9 @@
...
+  "leave_team_modal_public.desc": "**You will be removed from all {num_of_public_channels} public { num_of_public_channels,plural,one {channel} other {channels} } on this team**. Are you sure?",
```
Comment (jgilliam17) [inline i18n/en.json:3045]:
```suggestion
  "leave_team_modal_public.desc": "**You will be removed from {num_of_public_channels} public { num_of_public_channels,plural,one {channel} other {channels} } on this team.** Are you sure?",
```

File: i18n/en.json [hunk 5]
```diff
@@ -3041,7 +3041,9 @@
...
+  "leave_team_modal.desc": "**You will be removed from all {num_of_public_channels} public { num_of_public_channels,plural,one {channel} other {channels}} and {num_of_private_channels} private { num_of_private_channels,plural,one {channel} other {channels}} on this team**. If the team is private you will not be able to rejoin the team without an invitation. Are you sure?",
```
Comment (jgilliam17) [inline i18n/en.json:3046]:
```suggestion
  "leave_team_modal.desc": "**You will be removed from {num_of_public_channels} public { num_of_public_channels,plural,one {channel} other {channels}} and {num_of_private_channels} private { num_of_private_channels,plural,one {channel} other {channels}} on this team.** If the team is private you will not be able to rejoin the team without an invitation. Are you sure?",
```

**Review by jgilliam17 — CHANGES_REQUESTED** [review]
Thank you @esethna .
@sridhar02 - I added text suggestion edits to remove word `all` from modal text and made the punctuation mark `.` at the end of the first sentence also bold, so it is in the same typeface/style as the emphasized text. Thank you @justinegeffen for guidance.
- Can you please add a check if logged in user is a guest and they are leaving a team? We should show a deferent message in that case.

@andrewbrown00 Can you please confirm if this text change is acceptable? Thanks.

It should look something like this after the changes are applied.
Screen Shot 2020-10-19 at 12 25 04 PM

Sugested guest message text:
`**You will be removed from {num_of_public_channels} public { num_of_public_channels,plural,one {channel} other {channels}} and {num_of_private_channels} private { num_of_private_channels,plural,one {channel} other {channels}} on this team.** You will not be able to rejoin the team without an invitation. Are you sure?`

File: i18n/en.json (hunk 5 above)
Comment (sridhar02) [reply i18n/en.json:3072]: Resolved

File: i18n/en.json (hunk 4 above)
Comment (sridhar02) [reply i18n/en.json:3071]: Done

File: i18n/en.json (hunk 3 above)
Comment (sridhar02) [reply i18n/en.json:3070]: Done

File: components/leave_team_modal/leave_team_modal.tsx [hunk 6]
```diff
@@ -40,11 +46,47 @@ export default class LeaveTeamModal extends React.PureComponent<Props> {
...
 
     render() {
+        const numOfPublicChannels = this.props.publicChannels.length;
+        const numOfPrivateChannels = this.props.privateChannels.length;
+        let modalMessage;
+        if (numOfPublicChannels !== 0 && numOfPrivateChannels !== 0) {
+            modalMessage = (
+                <FormattedMarkdownMessage
+                    id='leave_team_modal.desc'
+                    defaultMessage='**You will be removed from all {num_of_public_channels} public { num_of_public_channels,plural,one {channel} other {channels} } and {num_of_private_channels} private {num_of_private_channels,one {channel} other {channels}} on this team**. If the team is private you will not be able to rejoin the team without an invitation. Are you sure?'
```
[the remark this answers is not in the document]
Comment (sridhar02) [reply components/leave_team_modal/leave_team_modal.tsx:64]: Done

File: components/leave_team_modal/leave_team_modal.tsx [hunk 7]
```diff
@@ -40,11 +46,47 @@ export default class LeaveTeamModal extends React.PureComponent<Props> {
...
+                    values={{
+                        num_of_public_channels: numOfPublicChannels,
+                        num_of_private_channels: numOfPrivateChannels,
+                    }}
+                />);
+        } else if (numOfPublicChannels === 0) {
+            modalMessage = (
+                <FormattedMarkdownMessage
+                    id='leave_team_modal_private.desc'
+                    defaultMessage='**You will be removed from all {num_of_private_channels} private {num_of_private_channels,one {channel} other {channels}} on this team**. If the team is private you will not be able to rejoin the team without an invitation. Are you sure?'
```
[the remark this answers is not in the document]
Comment (sridhar02) [reply components/leave_team_modal/leave_team_modal.tsx:74]: done

File: components/leave_team_modal/leave_team_modal.tsx [hunk 8]
```diff
@@ -40,11 +46,47 @@ export default class LeaveTeamModal extends React.PureComponent<Props> {
...
+                    values={{
+                        num_of_private_channels: numOfPrivateChannels,
+                    }}
+                />);
+        } else {
+            modalMessage = (
+                <FormattedMarkdownMessage
+                    id='leave_team_modal_public.desc'
+                    defaultMessage='**You will be removed from all {num_of_public_channels} public { num_of_public_channels,plural,one {channel} other {channels} } on this team**. Are you sure?'
```
[the remark this answers is not in the document]
Comment (sridhar02) [reply components/leave_team_modal/leave_team_modal.tsx:83]: done

File: components/leave_team_modal/__snapshots__/leave_team_modal.test.tsx.snap [hunk 9]
```diff
@@ -51,9 +51,14 @@ exports[`components/LeaveTeamModal should render the leave team model 1`] = `
     bsClass="modal-body"
     componentClass="div"
   >
-    <FormattedMessage
-      defaultMessage="You will be removed from all public and private channels.  If the team is private you will not be able to rejoin the team.  Are you sure?"
-      id="leave_team_modal.desc"
+    <injectIntl(FormattedMarkdownMessage)
+      defaultMessage="**You will be removed from all {num_of_private_channels} private {num_of_private_channels,one {channel} other {channels}} on this team**. If the team is private you will not be able to rejoin the team without an invitation. Are you sure?"
```
[the remark this answers is not in the document]
Comment (sridhar02) [reply components/leave_team_modal/__snapshots__/leave_team_modal.test.tsx.snap]: done

File: components/leave_team_modal/leave_team_modal.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (sridhar02) [reply components/leave_team_modal/leave_team_modal.tsx]: done

File: components/leave_team_modal/leave_team_modal.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (sridhar02) [reply components/leave_team_modal/leave_team_modal.tsx]: done

File: components/leave_team_modal/leave_team_modal.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (sridhar02) [reply components/leave_team_modal/leave_team_modal.tsx]: done

**Comment by sridhar02** [comment]
@jgilliam17 I have corrected the text and update the tests. can you review the PR?

**Comment by jgilliam17** [comment]
Thank you @sridhar02 
Updated text looks good.
- We still need to add a check - if user is a guest they should be seeing a slightly different message when leaving a team.
@matthewbirtch can you please advise if the suggested change/addition is acceptable. You can see my previous comment [above](https://github.com/mattermost/mattermost-webapp/pull/6572#pullrequestreview-511959806). Thank you.

File: i18n/en.json [hunk 10]
```diff
@@ -3067,7 +3067,9 @@
   "leave_private_channel_modal.leave": "Yes, leave channel",
   "leave_private_channel_modal.message": "Are you sure you wish to leave the private channel {channel}? You must be re-invited in order to re-join this channel in the future.",
   "leave_private_channel_modal.title": "Leave Private Channel {channel}",
-  "leave_team_modal.desc": "You will be removed from all public and private channels.  If the team is private you will not be able to rejoin the team.  Are you sure?",
+  "leave_team_modal_private.desc": "**You will be removed from {num_of_private_channels} private { num_of_private_channels,plural,one {channel} other {channels} } on this team.** If the team is private you will not be able to rejoin the team without an invitation. Are you sure?",
```
Comment (justinegeffen) [inline i18n/en.json:3070]:
```suggestion
  "leave_team_modal_private.desc": "**You will be removed from {num_of_private_channels} private { num_of_private_channels,plural,one {channel} other {channels} } on this team.** If the team is private you won't be able to rejoin it without an invitation from another team member. Are you sure?",
```

File: i18n/en.json [hunk 11]
```diff
@@ -3067,7 +3067,9 @@
...
+  "leave_team_modal_public.desc": "**You will be removed from {num_of_public_channels} public { num_of_public_channels,plural,one {channel} other {channels} } on this team.** Are you sure?",
+  "leave_team_modal.desc": "**You will be removed from {num_of_public_channels} public { num_of_public_channels,plural,one {channel} other {channels}} and {num_of_private_channels} private { num_of_private_channels,plural,one {channel} other {channels}} on this team.** If the team is private you will not be able to rejoin the team without an invitation. Are you sure?",
```
Comment (justinegeffen) [inline i18n/en.json:3072]:
```suggestion
  "leave_team_modal.desc": "**You will be removed from {num_of_public_channels} public { num_of_public_channels,plural,one {channel} other {channels}} and {num_of_private_channels} private { num_of_private_channels,plural,one {channel} other {channels}} on this team.** If the team is private you won't be able to rejoin it without an invitation from another team member. Are you sure?",
```

File: components/leave_team_modal/__snapshots__/leave_team_modal.test.tsx.snap [hunk 12]
```diff
@@ -51,9 +51,14 @@ exports[`components/LeaveTeamModal should render the leave team model 1`] = `
     bsClass="modal-body"
     componentClass="div"
   >
-    <FormattedMessage
-      defaultMessage="You will be removed from all public and private channels.  If the team is private you will not be able to rejoin the team.  Are you sure?"
-      id="leave_team_modal.desc"
+    <injectIntl(FormattedMarkdownMessage)
+      defaultMessage="**You will be removed from {num_of_private_channels} private {num_of_private_channels,one {channel} other {channels}} on this team.** If the team is private you will not be able to rejoin the team without an invitation. Are you sure?"
```
Comment (justinegeffen) [inline components/leave_team_modal/__snapshots__/leave_team_modal.test.tsx.snap:55]:
```suggestion
      defaultMessage="**You will be removed from {num_of_private_channels} private {num_of_private_channels,one {channel} other {channels}} on this team.** If the team is private you won't be able to rejoin it without an invitation from another team member. Are you sure?"
```

File: components/leave_team_modal/leave_team_modal.tsx [hunk 13]
```diff
@@ -40,11 +46,47 @@ export default class LeaveTeamModal extends React.PureComponent<Props> {
...
 
     render() {
+        const numOfPublicChannels = this.props.publicChannels.length;
+        const numOfPrivateChannels = this.props.privateChannels.length;
+        let modalMessage;
+        if (numOfPublicChannels !== 0 && numOfPrivateChannels !== 0) {
+            modalMessage = (
+                <FormattedMarkdownMessage
+                    id='leave_team_modal.desc'
+                    defaultMessage='**You will be removed from {num_of_public_channels} public { num_of_public_channels,plural,one {channel} other {channels} } and {num_of_private_channels} private {num_of_private_channels,one {channel} other {channels}} on this team.** If the team is private you will not be able to rejoin the team without an invitation. Are you sure?'
```
Comment (justinegeffen) [inline components/leave_team_modal/leave_team_modal.tsx:64]:
```suggestion
                    defaultMessage='**You will be removed from {num_of_public_channels} public { num_of_public_channels,plural,one {channel} other {channels} } and {num_of_private_channels} private {num_of_private_channels,one {channel} other {channels}} on this team.** If the team is private you won't be able to rejoin it without an invitation from another team member. Are you sure?'
```

File: components/leave_team_modal/leave_team_modal.tsx [hunk 14]
```diff
@@ -40,11 +46,47 @@ export default class LeaveTeamModal extends React.PureComponent<Props> {
...
+                    values={{
+                        num_of_public_channels: numOfPublicChannels,
+                        num_of_private_channels: numOfPrivateChannels,
+                    }}
+                />);
+        } else if (numOfPublicChannels === 0) {
+            modalMessage = (
+                <FormattedMarkdownMessage
+                    id='leave_team_modal_private.desc'
+                    defaultMessage='**You will be removed from {num_of_private_channels} private {num_of_private_channels,one {channel} other {channels}} on this team.** If the team is private you will not be able to rejoin the team without an invitation. Are you sure?'
```
Comment (justinegeffen) [inline components/leave_team_modal/leave_team_modal.tsx:74]:
```suggestion
                    defaultMessage='**You will be removed from {num_of_private_channels} private {num_of_private_channels,one {channel} other {channels}} on this team.** If the team is private you won't be able to rejoin it without an invitation from another team member. Are you sure?'
```

**Comment by justinegeffen** [comment]
I've suggested some rewording for the modals. @sridhar02, is it possible to confirm whether the changes have a negative impact on the modal appearance/format? Thank you!

**Comment by matthewbirtch** [comment]
> * @matthewbirtch can you please advise if the suggested change/addition is acceptable. You can see my previous comment [above](https://github.com/mattermost/mattermost-webapp/pull/6572#pullrequestreview-511959806). Thank you.

@jgilliam17 I'm good with your suggestions!

**Comment by jgilliam17** [comment]
Thank you for confirming @matthewbirtch 
@justinegeffen Thank you for the rewording suggestions. 
This is how the modal would look like with the changes, it still fits the format. @esethna thoughts? Thanks 
Screen Shot 2020-10-21 at 5 15 29 PM

**Review by esethna — APPROVED** [review]

**Comment by justinegeffen** [comment]
> Thank you for confirming @matthewbirtch
> @justinegeffen Thank you for the rewording suggestions.
> This is how the modal would look like with the changes, it still fits the format. @esethna thoughts? Thanks

FWIW, that looks good to me. :)

**Review by justinegeffen — APPROVED** [review]

**Comment by sridhar02** [comment]
@jgilliam17 just to reconfirm are there only changes in the rewordings or is there any need to add a condition and a new wording text.gor the modal.

**Comment by jgilliam17** [comment]
@sridhar02 Thank you for following up. 
Yes, beside the above suggested text update, we also need to add a condition to cover users that are guests and leaving a team
Guest should see this message when leaving the team:

`**You will be removed from {num_of_public_channels} public { num_of_public_channels,plural,one {channel} other {channels}} and {num_of_private_channels} private { num_of_private_channels,plural,one {channel} other {channels}} on this team.** You won't be able to rejoin it without an invitation from another team member. Are you sure?`

Let me know if you need more information. Thanks 🙂

File: i18n/en.json (hunk 10 above)
Comment (sridhar02) [reply i18n/en.json:3070]: Done

File: i18n/en.json (hunk 11 above)
Comment (sridhar02) [reply i18n/en.json:3072]: done

File: components/leave_team_modal/__snapshots__/leave_team_modal.test.tsx.snap (hunk 12 above)
Comment (sridhar02) [reply components/leave_team_modal/__snapshots__/leave_team_modal.test.tsx.snap:55]: done

File: components/leave_team_modal/leave_team_modal.tsx (hunk 13 above)
Comment (sridhar02) [reply components/leave_team_modal/leave_team_modal.tsx:64]: done

File: components/leave_team_modal/leave_team_modal.tsx (hunk 14 above)
Comment (sridhar02) [reply components/leave_team_modal/leave_team_modal.tsx:74]: done

**Comment by sridhar02** [comment]
@sudheerDev @hmhealey can you point me how to add the leave team modal message for a guest or how to check for a guest?

**Comment by hmhealey** [comment]
I believe guest accounts are an Enterprise Edition feature. They should be available on the test server on this PR, but if you want to test them out locally, you could fake a guest account by modifying [`Utils.isGuest`](https://github.com/mattermost/mattermost-webapp/blob/master/utils/utils.jsx#L169) to always return true for one or more users. Doing that might have some side effects, but it would let you simulate what a guest user sees in that modal

**Comment by hmhealey** [comment]
Re-adding dev review so I can double check the changes after the text is sorted out.

**Comment by sridhar02** [comment]
@hmhealey @sudheerDev  can you review this PR? Added some changes to show the new modal message for the guest. 

@jgilliam17 @justinegeffen can you recheck the message for guest and message wordings from your end also?

**Review by hmhealey — APPROVED** [review]
Thanks for keeping up with all the wording changes as we figure this out. It still looks good to me code-wise.

File: i18n/en.json [hunk 15]
```diff
@@ -3070,7 +3070,10 @@
   "leave_private_channel_modal.leave": "Yes, leave channel",
   "leave_private_channel_modal.message": "Are you sure you wish to leave the private channel {channel}? You must be re-invited in order to re-join this channel in the future.",
   "leave_private_channel_modal.title": "Leave Private Channel {channel}",
-  "leave_team_modal.desc": "You will be removed from all public and private channels.  If the team is private you will not be able to rejoin the team.  Are you sure?",
+  "leave_team_modal_guest.desc": "**You will be removed from {num_of_public_channels} public { num_of_public_channels,plural,one {channel} other {channels}} and {num_of_private_channels} private { num_of_private_channels,plural,one {channel} other {channels}} on this team.** You won't be able to rejoin it without an invitation from another team member. Are you sure?",
```
Comment (justinegeffen) [inline i18n/en.json:3073]:
```suggestion
  "leave_team_modal_guest.desc": "**You will be removed from {num_of_public_channels} public { num_of_public_channels,plural,one {channel} other {channels}} and {num_of_private_channels} private { num_of_private_channels,plural,one {channel} other {channels}} on this team.** If the team is private you won't be able to rejoin it without an invitation from another team member. Are you sure?",
```

File: i18n/en.json [hunk 16]
```diff
@@ -3070,7 +3070,10 @@
...
+  "leave_team_modal_private.desc": "**You will be removed from {num_of_private_channels} private { num_of_private_channels,plural,one {channel} other {channels} } on this team.** If the team is private you won't be able to rejoin it the team without an invitation from another team member. Are you sure?",
```
Comment (justinegeffen) [inline i18n/en.json:3074]:
```suggestion
  "leave_team_modal_private.desc": "**You will be removed from {num_of_private_channels} private { num_of_private_channels,plural,one {channel} other {channels} } on this team.** If the team is private you won't be able to rejoin it without an invitation from another team member. Are you sure?",
```

**Review by justinegeffen — APPROVED** [review]

**Comment by justinegeffen** [comment]
One minor fix, but otherwise LGTM, thank you!!

File: i18n/en.json (hunk 16 above)
Comment (sridhar02) [reply i18n/en.json:3074]: @justinegeffen it is showing wrong from the files section, I have opened the actual file it is correct there, can you recheck this by opening this https://github.com/mattermost/mattermost-webapp/blob/2e2502234d7b1e7340de141424d5670e2c865f10/i18n/en.json#L3073?

**Comment by jgilliam17** [comment]
Thank you @sridhar02 for all the great changes
one issue for guest - if there are no public or private channels that the guest user is a member of we should not display 0 count in the modal
Screen Shot 2020-10-27 at 9 28 36 AM
Screen Shot 2020-10-27 at 9 27 11 AM

**Comment by sridhar02** [comment]
@jgilliam17 what will be the texts for guests will those same these?

https://github.com/mattermost/mattermost-webapp/blob/2e2502234d7b1e7340de141424d5670e2c865f10/i18n/en.json#L3074

https://github.com/mattermost/mattermost-webapp/blob/2e2502234d7b1e7340de141424d5670e2c865f10/i18n/en.json#L3075

**Comment by jgilliam17** [comment]
@sridhar02
Here are few points 
- The guest message should not dependent on the team type. Use same message when guest leaves either private or public team
- We should count the private and public channels on the team and display that information, when count is zero, omit that part of message.
Screen Shot 2020-10-27 at 9 28 36 AM

The rest of the message remains the same: `You won't be able to rejoin it without an invitation from another team member. Are you sure?`

Don't hesitate to ping me here if I misunderstood what you were asking about 🙂

**Comment by sridhar02** [comment]
@hmhealey @sudheerDev  https://github.com/mattermost/mattermost-webapp/blob/2196c6b5635de31e6bd0434faa87f988f4fac803/components/leave_team_modal/leave_team_modal.tsx#L65 currently they are 5 conditional statements and it becoming huge can you suggest any better way to write these conditions?

@jgilliam17 can you recheck the PR I have modified the messages for the guest?

**Comment by jgilliam17** [comment]
@sridhar02 It doesn't appear like all the channels are checked

Screen Shot 2020-10-28 at 8 07 59 AM

**Comment by sridhar02** [comment]
@jgilliam17 can you recheck now, I have slight written wrong conditions earlier now I corrected them?

**Comment by hmhealey** [comment]
The code looks a lot better with not repeating the `<FormattedMessage` part over an over. The only other way I could see making it simpler would be breaking up the three sentences since the first one only changes based on the number of channels, the second changes on whether or not the user is a guest, and the third always stays the same, but I'd be fine leaving it as-is since concatenating translated text can lead to problems if we're not careful.

**Comment by jgilliam17** [comment]
@sridhar02 there's still an issue with conditions for guest message

Leaving team + only public channels > counts 0 private 
Leaving team + only private channels > counts 0 public
Leaving team + both private and public channels > counts both correctly 
Screen Shot 2020-10-28 at 2 11 04 PM
Screen Shot 2020-10-28 at 2 10 11 PM
Screen Shot 2020-10-28 at 2 10 44 PM

**Comment by sridhar02** [comment]
@jgilliam17 sorry for the earlier bugs, I should have checked them and then inform to you, earlier I don't know how to log in as a guest so I haven't checked it my self now went to the through the manual to create a guest and created a guest and checked it my self, I think those guest messages are fixed now, can you recheck them?

**Review by jgilliam17 — APPROVED** [review]
No problem at all @sridhar02! 
Re-tested, looks good to merge.  
- Verified help text messages when user is leaving the team look good (conditions and text).

Thank you so much for your contribution and for keeping up with all the back and forth while we were trying to figure out all the changes.

**Comment by sridhar02** [comment]
@hmhealey can you merge this PR, I think all the reviews are completed for this PR?

**Review by sudheerDev — APPROVED** [review]
Approving again after the changes

**Comment by sudheerDev** [comment]
@justinegeffen Hey, Are you good with the change requests and changes done?

**Review by justinegeffen — APPROVED** [review]

**Comment by justinegeffen** [comment]
> @justinegeffen Hey, Are you good with the change requests and changes done?

I am, thank you for the ping! :)

**Pull request merged**